src/main/groovy/pl/allegro/tech/build/axion/release/domain/SnapshotDependenciesChecker.groovy

```diff
@@ -0,0 +1,27 @@
+package pl.allegro.tech.build.axion.release.domain
+
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Dependency
+
+class SnapshotDependenciesChecker {
+
+    Collection<String> snapshotVersions(Project project) {
+        def projectVersions = project.allprojects.collect {toFullVersion(it)}
+        def allDependenciesVersions = project.allprojects.collect {
+            it.configurations.collect { config ->
+                config.allDependencies.findAll {isSnapshot(it)}.collect {toFullVersion(it)}
+
+            }
+        }.flatten().unique()
+        allDependenciesVersions.removeAll(projectVersions)
+        return allDependenciesVersions
+    }
+
+    boolean isSnapshot(Dependency dependency) {
+        dependency.version?.endsWith("-SNAPSHOT")
+    }
+
+    String toFullVersion(it) {
+        "${it.group}:${it.name}:${it.version}"
```

Comment:
I think that @vbuell rebased the code in the meantime (hmm, usually I was able to get "outdated diff", but in that case it seems to be not possible outside a small diff on the conversation page), from something like:
```
def projectVersions = ...
def allDependenciesVersions = ...
```
I also didn't like `String toFullVersion(it) {}` and it thought that `Project` could be used there, but it seems that there could be also a `Dependency` instance which has nothing common with `Project` (regarding a class hierarchy).